### DISCUSSION TITLE
no discrimination for inline-math and math-block?

### DISCUSSION THREAD

**Discussion by shi-yan** [wooorm/markdown-rs#100] [Q&A] [answered]
In my markdown, I have both inline-math (between single dollar signs) and math-blocks (between double dollar signs).

But when I render the html, they all become inline-maths. I want to maintain the inline or block type, because I want to format them differently, for math block, I want to center the equation:

<img width="814" alt="Screenshot 2024-01-13 at 11 06 25 AM" src="https://github.com/wooorm/markdown-rs/assets/326807/285631f6-bb35-4330-a3e0-0495c03f77f9">

**Comment by wooorm** [comment]
It follows the same rules as code in markdown. You can use three ticks for "inline" and "block" too. So that isn't it. It's whether you use enters or not.

Please post code with your questions. I can't look into your mind.

**Reply by shi-yan** [reply]
here is the markdown code I have:

```md
When presented with a data-point $x_0$ sampled from the actual data distribution $q(x) (x_0 \sim q(x))$, a forward diffusion process can be defined. This process involves adding Gaussian noise with variance $\beta_{t}$ to $x_{t-1}$, thereby generating a new sample $x_{t}$ governed by the distribution $q(x_{t} \mid x _{t-1} )$. This formulation of the diffusion process can be articulated as follows:

$$
  q(x_{t}|x_{t-1})= \mathcal{N}(x_{t}; \mu _{t} = \sqrt {1-\beta _{t}}x _{t-1}, \Sigma_{t} = \beta _{t} I)
$$

```

everything is rendered as inline-math

**Reply by wooorm** [reply]
What is "rendered as"?

This project turns markdown into html or ASTs. Perhaps you need some CSS or JavaScript to render things the way you want?

**Reply by shi-yan** [reply]
they are rendered as `<code class="language-math math-inline">xxx</code>` for both single dollar and double dollar clauses. I also checked the ast, the nodes are all inline-math nodes, instead of the math node for the double dollar case. I think this behavior is wrong.

**Reply by wooorm** [reply]
The tests clearly show that is not the case? https://github.com/wooorm/markdown-rs/blob/60db8e5896be05d23137a6bdb806e63519171f9e/tests/math_flow.rs#L31

**Reply by shi-yan** [reply]
here is a minimal reproducible example:

```rust
use markdown::{Constructs, Options, ParseOptions};

fn main() {
    let mut options = Options::gfm();
    options.parse.constructs.math_text = true;
    options.parse.constructs.frontmatter = true;

    let markdown = "test $x_0$ \n\
\n\
    $$\n\
    x_0 \n\
    $$";

    let ast = markdown::to_mdast(&markdown, &options.parse).unwrap();

    println!("{:?}", ast);
}

```

the above produces the following ast:

```
Root { children: [Paragraph { children: [Text { value: "test ", position: Some(1:1-1:6 (0-5)) }, InlineMath { value: "x_0", position: Some(1:6-1:11 (5-10)) }], position: Some(1:1-1:12 (0-11)) }, Paragraph { children: [InlineMath { value: "\nx_0 \n", position: Some(3:1-5:3 (13-23)) }], position: Some(3:1-5:3 (13-23)) }], position: Some(1:1-5:3 (0-23)) }
```

**Reply by shi-yan** [reply, marked as answer]
sorry, I realized that I need to enable `math_flow`

**Reply by wooorm** [reply]
You specifically enable only `math_text`. So you get only math text...

`math_flow = true`.
